From b2226d37a00862c791bcf1401b2e76adae43a1e8 Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@rvc.renesas.com>
Date: Fri, 26 Feb 2016 18:01:50 +0700
Subject: [PATCH 230/909] PCI: pcie-rcar: Add empty suspend/resume handlers

commit c89f8288cc02d4ffcaecd5451a85d81b21f56031 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Suspend To RAM should be supported for PCIe.
Right now, just empty callbacks.
Detail implementation will be introduced later.

Signed-off-by: Hien Dang <hien.dang.eb@rvc.renesas.com>
Signed-off-by: Tho Vu <tho.vu.wh@rvc.renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/pcie-rcar.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/drivers/pci/controller/pcie-rcar.c b/drivers/pci/controller/pcie-rcar.c
index 223b42d05cc5..fdb38b2004c1 100644
--- a/drivers/pci/controller/pcie-rcar.c
+++ b/drivers/pci/controller/pcie-rcar.c
@@ -1245,9 +1245,32 @@ static int rcar_pcie_probe(struct platform_device *pdev)
 	return err;
 }
 
+#ifdef CONFIG_PM_SLEEP
+static int rcar_pcie_suspend(struct device *dev)
+{
+	/* Empty functino for now */
+	return 0;
+}
+
+static int rcar_pcie_resume(struct device *dev)
+{
+	/* Empty function for now */
+	return 0;
+}
+
+static SIMPLE_DEV_PM_OPS(rcar_pcie_pm_ops,
+			rcar_pcie_suspend,
+			rcar_pcie_resume);
+
+#define DEV_PM_OPS (&rcar_pcie_pm_ops)
+#else
+#define DEV_PM_OPS NULL
+#endif /* CONFIG_PM_SLEEP */
+
 static struct platform_driver rcar_pcie_driver = {
 	.driver = {
 		.name = "rcar-pcie",
+		.pm	= DEV_PM_OPS,
 		.of_match_table = rcar_pcie_of_match,
 		.suppress_bind_attrs = true,
 	},
-- 
2.17.1

